### PR TITLE
RFR: Rule update 500 no pack fix

### PR DESCRIPTION
**Pull request by lakshmi-kannan**

### CODE REVIEW

File: st2common/st2common/models/api/rule.py [hunk 1]
```diff
@@ -162,7 +163,7 @@ def to_model(cls, rule):
 
         trigger = reference.get_str_resource_ref_from_model(trigger_db)
         criteria = dict(getattr(rule, 'criteria', {}))
-        pack = str(rule.pack)
+        pack = getattr(rule, 'pack', DEFAULT_PACK_NAME)
```
Comment (manasdk) [inline st2common/st2common/models/api/rule.py]: Should we make `pack` required? 

OTH setting pack to default might not be the right approach either.

Comment (dzimine) [inline st2common/st2common/models/api/rule.py]: Isn't action defaults to the pack which contains it? Which was handy for creating them in one pack (e.g. default) and moving to another without changing the metadata files. 
I think we should apply the same logic to all types of content.

Comment (lakshmi-kannan) [inline st2common/st2common/models/api/rule.py]: +1 to what dzimine says. It should be consistent. We can either make `pack` required everywhere or optional everywhere. 

>Which was handy for creating them in one pack (e.g. default) and moving to another without changing the metadata files.

There is a problem that we don't clean up the one in default.

Comment (Kami) [inline st2common/st2common/models/api/rule.py]: Yeah, we have the same logic in actions controller right now (defaulting to "default" pack if `pack` is not provided), but I would prefer us to eventually move to `pack` being required.

Comment (Kami) [inline st2common/st2common/models/api/rule.py]: It's also worth clarifying, that as far as I know, this only applies to the rules which were created using the API before we added `pack` field to the `Rule` model.

When we added `pack` field, we also added a change which sets `pack` to a default value if the value is not provided in the API request payload.

**Comment by lakshmi-kannan** [comment]
I am going to rage merge this.

**Pull request merged**